src/gsi/da_adBuffer.f90

```diff
@@ -0,0 +1,1358 @@
+      INTEGER adl4ibuf,adl4ilbuf
+      LOGICAL adl4inlbuf
+      COMMON /adl4fbuf/adl4buf,adl4lbuf,                                &
+     &       adl4ibuf,adl4ilbuf,adl4inlbuf
+!
+      IF (adl4ilbuf.ne.-1) THEN
+         adl4ilbuf = -1
+         adl4inlbuf = .FALSE.
+      ENDIF
+      IF (adl4ibuf.le.1) THEN
+         CALL POPBOOLEANARRAY(adl4buf, 512)
+         adl4ibuf = 512
+         x = adl4buf(512)
+      ELSE
+         adl4ibuf = adl4ibuf-1
+         x = adl4buf(adl4ibuf)
+      ENDIF
+      END
+
+!======================= INTEGER*4 =========================:
+      BLOCK DATA INTEGERS4
+      INTEGER*4 adi4buf(512), adi4lbuf(512)
+      INTEGER adi4ibuf,adi4ilbuf
+      LOGICAL adi4inlbuf
+      COMMON /adi4fbuf/adi4buf,adi4lbuf,                                &
+     &       adi4ibuf,adi4ilbuf,adi4inlbuf
+      DATA adi4ibuf/1/
+      DATA adi4ilbuf/-1/
+      DATA adi4inlbuf/.FALSE./
+      END
+
+      SUBROUTINE PUSHINTEGER4(x)
+      INTEGER*4 x, adi4buf(512), adi4lbuf(512)
+      INTEGER adi4ibuf,adi4ilbuf
+      LOGICAL adi4inlbuf
+      COMMON /adi4fbuf/adi4buf,adi4lbuf,                                &
+     &       adi4ibuf,adi4ilbuf,adi4inlbuf
+!
+      call addftraffic(4)
+      IF (adi4ilbuf.ne.-1) THEN
+         adi4ilbuf = -1
+         adi4inlbuf = .FALSE.
+      ENDIF
+      IF (adi4ibuf.ge.512) THEN
+         adi4buf(512) = x
+         CALL PUSHINTEGER4ARRAY(adi4buf, 512)
+      call addftraffic(-2048)
+         adi4ibuf = 1
+      ELSE
+         adi4buf(adi4ibuf) = x
+         adi4ibuf = adi4ibuf+1
+      ENDIF
+      END
+
+      SUBROUTINE LOOKINTEGER4(x)
+      INTEGER*4 x, adi4buf(512), adi4lbuf(512)
+      INTEGER adi4ibuf,adi4ilbuf
+      LOGICAL adi4inlbuf
+      COMMON /adi4fbuf/adi4buf,adi4lbuf,                                &
+     &       adi4ibuf,adi4ilbuf,adi4inlbuf
+!
+      IF (adi4ilbuf.eq.-1) THEN
+         adi4ilbuf=adi4ibuf
+         call RESETADLOOKSTACK()
+      ENDIF
+      IF (adi4ilbuf.le.1) THEN
+         CALL LOOKINTEGER4ARRAY(adi4lbuf, 512)
+         adi4inlbuf = .TRUE.
+         adi4ilbuf = 512
+         x = adi4lbuf(512)
+      ELSE
+         adi4ilbuf = adi4ilbuf-1
+         if (adi4inlbuf) THEN
+            x = adi4lbuf(adi4ilbuf)
+         ELSE
+            x = adi4buf(adi4ilbuf)
+         ENDIF
+      ENDIF
+      END
+
+      SUBROUTINE POPINTEGER4(x)
+      INTEGER*4 x, adi4buf(512), adi4lbuf(512)
+      INTEGER adi4ibuf,adi4ilbuf
+      LOGICAL adi4inlbuf
+      COMMON /adi4fbuf/adi4buf,adi4lbuf,                                &
+     &       adi4ibuf,adi4ilbuf,adi4inlbuf
+!
+      IF (adi4ilbuf.ne.-1) THEN
+         adi4ilbuf = -1
+         adi4inlbuf = .FALSE.
+      ENDIF
+      IF (adi4ibuf.le.1) THEN
+         CALL POPINTEGER4ARRAY(adi4buf, 512)
+         adi4ibuf = 512
+         x = adi4buf(512)
+      ELSE
+         adi4ibuf = adi4ibuf-1
+         x = adi4buf(adi4ibuf)
+      ENDIF
+      END
+
+!======================= INTEGER*8 =========================
```

Comment:
Please clean up the code, remove unnecessary lines, only put the proper documentation in the head.

Reply:
okay, removed line 249